ocaml/parsing/builtin_attributes.mli

```diff
@@ -207,24 +207,34 @@ val jkind : Parsetree.attributes -> jkind_attribute Location.loc option
     ignored. **)
 val error_message_attr : Parsetree.attributes -> string option
 
-(** [get_payload] is a helper function for working with attribute payloads.
-    Given a payload that consists of a structure containing a single
-    [Pstr_eval], it will call the given function on the inner expression.
-    Otherwise, it returns [Error ()]. *)
-val get_payload :
-  (Parsetree.expression -> ('a, unit) Result.t) -> Parsetree.payload ->
-  ('a, unit) Result.t
-
-(** [get_optional_payload] is like [get_payload], except that in the case that
-    the payload consists of an empty structure, it returns [Ok None] rather than
-    [Error ()] . *)
-val get_optional_payload :
-  (Parsetree.expression -> ('a, unit) Result.t) -> Parsetree.payload ->
-  ('a option, unit) Result.t
-
-(** [get_id_from_exp] extracts the string from an identifier expression.
-    Suitable for use with [get_payload]. *)
-val get_id_from_exp : Parsetree.expression -> (string, unit) Result.t
+(** [get_int_payload] is a helper for working with attribute payloads.
+    Given a payload that consist of a structure containing exactly
+    {[
+      PStr [
+        {pstr_desc =
+           Pstr_eval (Pexp_constant (Pconst_integer(i, None)), [])
+        }
+      ]
+    ]}
+    it returns [i].
+  *)
+val get_int_payload : Parsetree.payload -> (int, unit) Result.t
+
+(** [get_optional_bool_payload] is a helper for working with attribute payloads.
+    It behaves like [get_int_payload], except that it looks for a boolean
+    constant rather than an int constant, and returns [None] rather than [Error]
+    if the payload is empty. *)
+val get_optional_bool_payload :
+    Parsetree.payload -> (bool option, unit) Result.t
+
+(** [parse_id_payload] is a helper for parsing information from an identifier
+    attribute payload. If the given payload consists of a single identifier,
+    that identifier is looked up in the association list.  The result is
+    returned, if it exists.  The [empty] value is used if the payload is empty.
+    Otherwise the [default] value is used an a warning is issued. *)
+val parse_id_payload :
+  string -> Location.t -> default:'a -> empty:'a -> (string * 'a) list ->
+  Parsetree.payload -> 'a
```

Comment:
`get_int_payload` and `get_optional_bool_payload` have clear and simple documentation and types. I think we should keep them in `builtin_attributes.ml`. `parse_id_payload` is weirdly complex, and it seems less compelling to me to hoist it out to a more fundamental attribute-parsing library like `builtin_attributes.ml`. Could we keep it in `translattribute.ml`, closer to its use sites?

Reply:
`parse_ids_payload` (with an `s`!) is needed in this file for `parse_property_attribute`.  I moved `parse_id_payload` (without an `s`) because it's very similar to the `ids` version and needs similar helper functions that I otherwise have to expose in the `mli` here for only this purpose.  It seemed better to me to put the very similar `id` and `ids` versions together and avoid exposing the helpers, even though I agree `id` is a little weird in the mli.

I'm not super happy with this, but there's an explanation of the tradeoffs, anyway.  If you still think it's better to move it, I will.

Reply:
Augh, I fold. I don't think it's better to move it. I will make some suggestions though:
  * "identifier attribute payload" is an unfamiliar term. Maybe "attribute whose payload is an identifier"?
  * `parse_optional_id_payload`? It is `optional`, that's what `empty` is for.
  * Get rid of `default` and instead return `Error ()`. `default` and `empty` are just too easy to conceptually confuse at the moment.

Reply:
Done, thanks.